main.tf

```diff
@@ -121,7 +121,8 @@ resource "aws_security_group_rule" "ingress_cidr_blocks" {
 }
 
 data "aws_ami" "eks_worker" {
-  count       = "${var.enabled == "true" && var.image_id == "" ? 1 : 0}"
+  count = "${var.enabled == "true" && var.use_custom_image_id == "true" ? 1 : 0}"
```

Comment:
```suggestion
  count = "${var.enabled == "true" && var.use_custom_image_id == "false" ? 1 : 0}"
```

Reply:
@dmitriy-lukyanchikov  this is wrong logic, please update

Reply:
we should lookup the AMI only when `var.use_custom_image_id == "false"`

Reply:
totally miss this thank you